File: DataSets/DataSet-New/LeetCodeTasks - Debug/Fixed/3000_maximum-area-of-longest-diagonal-rectangle/solutions/java/solution.java
class Solution {

    public int areaOfMaxDiagonal(int[][] dimensions) {

        double maxDiagonal = 0;
        int maxArea = 0;

        for (int[] dimension : dimensions) {
            int length = dimension[0];
            int width = dimension[1];

            double diagonal = Math.sqrt(length * length + width * width); // Fixed formula to correctly calculate the diagonal
            int area = length * width;

            if (diagonal > maxDiagonal || (diagonal == maxDiagonal && area > maxArea)) { // Fixed the condition to correctly handle both checks
                maxDiagonal = diagonal;
                maxArea = area;
            }
        }

        return maxArea;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Debug/Fixed/3000_maximum-area-of-longest-diagonal-rectangle/solutions/c/solution.c
#include <math.h>

int areaOfMaxDiagonal(int** dimensions, int dimensionsSize, int* dimensionsColSize) {
    double maxDiagonal = 0.0;
    int maxArea = 0;

    for (int i = 0; i < dimensionsSize; i++) {
        int length = dimensions[i][0];
        int width = dimensions[i][1];
        double diagonal = sqrt(length * length + width * width);
        int area = length * width;

        if (diagonal > maxDiagonal) {
            maxDiagonal = diagonal;
            maxArea = area;
        } else if (diagonal == maxDiagonal && area > maxArea) {
            maxArea = area;
        }
    }

    return maxArea;
}
